File: solution/java/week06/exercise2/src/main/java/report/PdfReportGenerator.java
package report;

import java.util.List;

public class PdfReportGenerator implements CanGenerateReport {
    @Override
    public String generateReport(List<ReportData> data) {
        StringBuilder sb = new StringBuilder("Starting PDF Report Generation...\nPDF Report Title: Comprehensive Data Report\n------------------------------------------------\n");
        data.forEach(d -> sb.append("Data ID: ").append(d.getId()).append(" | ").append("Data Value: ").append(d.getValue()).append(" | ").append("Description: ").append(d.getDescription()).append("\n"));
        sb.append("------------------------------------------------\nPDF Report Generated Successfully.");
        return sb.toString();
    }
}

File: solution/ts/week06-legacy-code/exercise2/src/pdfReportGenerator.ts
import {CanGenerateReport} from "./canGenerateReport";
import {ReportData} from "./reportData";

export class PdfReportGenerator implements CanGenerateReport {
    generateReport(data: ReportData[]): string {
        let report = "Starting PDF Report Generation...\nPDF Report Title: Comprehensive Data Report\n------------------------------------------------\n";
        data.forEach(d => {
            report += `Data ID: ${d.id} | Data Value: ${d.value} | Description: ${d.description}\n`;
        });
        report += "------------------------------------------------\nPDF Report Generated Successfully.";
        return report;
    }
}
